peleffy/tests/test_solvent.py

```diff
@@ -6,15 +6,85 @@
-from peleffy.solvent import OPLSOBC
+from peleffy.forcefield import OPLS2005ForceField, OpenForceField
+from peleffy.solvent import OPLSOBC, OBC2
 
 
 class TestSolvent(object):
     """
     It contains all the tests that validate the solvent-template generator.
     """
 
+    def test_single_topology(self):
+        """
+        It tests the class that generates a OpenFFCompatibleSolvent object for
+        a single topology.
+        """
+        from .utils import compare_dicts
+        import json
+
+        TEMPLATE_PARAMS_MAL = get_data_file_path('tests/ligandParams_MAL.txt')
+
+        # Loads the  molecule
+        molecule = Molecule(path=get_data_file_path('ligands/malonate.pdb'),
+                            tag='MAL')
+
+        # Sets forcefield and parameterizes it
+        ff = OpenForceField('openff_unconstrained-1.2.1.offxml')
+        parameters = ff.parameterize(molecule, charge_method='gasteiger')
+
+        # Initializes topology
+        topology = Topology(molecule, parameters)
+
+        # Initializes solvent and gets parameters file
+        solvent = OBC2(topology)
+        solvent_dict = solvent.to_dict()
+
+        # Loads reference dict from template
+        with open(TEMPLATE_PARAMS_MAL, 'r') as f:
+            reference_dict = json.load(f)
+
+        # Compare the output parameters dict with the reference parameters
+        compare_dicts(reference_dict, solvent_dict)
+
+    def test_multiple_topologies(self):
+        """
+        It tests the class that generates a OpenFFCompatibleSolvent object for
+        multiple topologies.
+        """
+        from .utils import compare_dicts, merge_dicts
+
+        # Path to multiple non standard residues
+        pdb_path_MAL = get_data_file_path('ligands/malonate.pdb')
+        pdb_path_MET = get_data_file_path('ligands/methane.pdb')
+
+        # Force Field to parameterize the molecules
+        ff = OpenForceField('openff_unconstrained-1.2.1.offxml')
+
+        # Topology of malonate
+        mol_MAL = Molecule(path=pdb_path_MAL, tag='MAL')
+        parameters_MAL = ff.parameterize(mol_MAL, charge_method='gasteiger')
+        topology_MAL = Topology(mol_MAL, parameters_MAL)
+
+        # Topology of methane
+        mol_MET = Molecule(path=pdb_path_MET, tag='MET')
+        parameters_MET = ff.parameterize(mol_MET, charge_method='gasteiger')
+        topology_MET = Topology(mol_MET, parameters_MET)
+
+        # List containing both topologies
+        topologies = [topology_MAL, topology_MET]
+
+        # Generate the Solvent parameters dictionaries
+        solvent_MAL_dict = OBC2(topology_MAL).to_dict()
+        solvent_MET_dict = OBC2(topology_MET).to_dict()
+        solvent_dict = OBC2(topologies).to_dict()
+
+        # Check that merging both single topology dicitionaries we obtain the
+        # same dictionary that using multiple topologies
+        compare_dicts(merge_dicts(solvent_MAL_dict['SolventParameters'],
```

Comment:
This is a nice way to validate the `Solvent` object with multiple topologies. But I am missing one test that validates the output file that results from a `Solvent` with multiple topologies. Something similar that is done in the `test_single_topology` test comparing the outcome of the `Solvent` writer with a reference file.